pkg/kubelet/network/dns/dns_test.go

```diff
@@ -445,8 +468,14 @@ func TestGetPodDNS(t *testing.T) {
 		t.Errorf("expected nameserver %s, got %v", clusterNS, options[0].DNS[0])
 	}
 	expLength := len(options[1].DNSSearch) + 3
-	if expLength > 6 {
-		expLength = 6
+
+	maxDNSSearchPaths := validation.MaxDNSSearchPathsLegacy
+	if utilfeature.DefaultFeatureGate.Enabled(features.ExpandedDNSConfig) {
```

Comment:
You can't use the gate like this in a test - you need to run the test cases both with and without it